### ISSUE TITLE
Remotely Resident Agents EXE is broken with latest update

### ISSUE THREAD

**Issue by tim1014** [immense/Remotely#735]
**Describe the bug**
Remotely Resident Agents is broken with latest update

**To Reproduce**
1. Download Resident Agents at Download page >> Windows Installer (x64/x86)
2. The EXE will never runs.

**Remotely Version**
Server (can be found on about page): v2023.09.11.1010 Release
Agent (can be found in device card): N/A

**Expected Behavior**
The EXE should run as expected.

**Screenshots**
![image](https://github.com/immense/Remotely/assets/19739878/5ce85065-9a2e-41e7-8c7a-d76e79227e6b)

**Desktop (please complete the following information):**
 - OS: Windows 10 22H2
 - Browser: Firefox 119.0a1
 
 I have tried to install and run docker on a clean machine with the latest image, same thing happens.

**Comment by nittanygeek** [comment]
Running into the same issue.  Windows 11 (64 bit), using the docker container on version 2023.09.11.1010.  Same error message as in screenshot.  Manually running via CLI results in:

`ResourceUnavailable: Program 'Remotely_Installer.exe' failed to run: An error occurred trying to start process 'C:\Users\[redacted]\Downloads\Remotely\Remotely_Installer.exe' with working directory 'C:\Users\[redacted]\Downloads\Remotely'. The specified executable is not a valid application for this OS platform.At line:1 char:1`

**Comment by xFrieDSpuDx** [comment]
I am also having issues installing the Resident Agent, this time on Windows 11 Pro. When run the installer silently fails. In the event viewer logs I get the following two entries.
**Entry One**
Faulting application name: Remotely_Installer.exe, version: 0.0.0.0, time stamp: 0x00000000
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x641eff68
Faulting process id: 0x2e9c
Faulting application start time: 0x01d9e628833546a6
Faulting application path: C:\Users\username\Downloads\Remotely_Installer.exe
Faulting module path: unknown
Report Id: a9d9c976-3337-4cc1-9123-8133872269e5
Faulting package full name: 
Faulting package-relative application ID: 

**Entry Two**
Fault bucket 1572129834046190372, type 1
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Remotely_Installer.exe
P2: 0.0.0.0
P3: 00000000
P4: StackHash_0000
P5: 0.0.0.0
P6: 00000000
P7: c0000005
P8: PCH_03
P9: 
P10: 

Attached files:
\\?\C:\Users\username\AppData\Local\Temp\WER.5df30e28-69f7-4cb8-a115-26b3b21b1d44.tmp.WERDataCollectionStatus.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.33f13da2-19dc-40d9-b286-54c8ba9cc5ba.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.8af6edf3-f84c-43d0-8999-7399950a76ae.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.7698190e-b689-42a2-b2bb-0e2267bf869f.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER.c56ef8b9-f532-463a-a801-f308c904253c.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Remotely_Install_35bfee27f63c6bb6eabf78e5f6314a88737469c9_f08573da_710e56c2-6719-4a10-ad51-41e7fa635926

Analysis symbol: 
Rechecking for solution: 0
Report Id: a9d9c976-3337-4cc1-9123-8133872269e5
Report Status: 268435456
Hashed bucket: 939010657ab4c70f55d153c12b1df324
Cab Guid: 0

**Comment by nittanygeek** [comment]
I found a temporary workaround.  The button in Downloads is linking to https://redacted/API/ClientDownloads/WindowsInstaller, however after poking around the code a bit I found the actual Remotely_Installer.exe is located in https://redacted/Content/Remotely_Installer.exe

I was able to download the installer directly and then Right Click > Properties and unblock the file before I could run it.

Hope this helps.

**Comment by xFrieDSpuDx** [comment]
@nittanygeek phenomenal find! I can confirm that your temporary workaround works for me.

**Comment by nittanygeek** [comment]
> @nittanygeek phenomenal find! I can confirm that your temporary workaround works for me.

Thanks!  I'm still running into all kinds of problems with SSL yet because I'm using Nginx Proxy Manager (which worked fine before Immense bought out the code).  Seems the only fix there is to switch to Caddy, but that's a separate issue on it's own.

**Comment by xFrieDSpuDx** [comment]
If it makes you feel better, I have no issues with SSL, offloading them with HAProxy, but I can't get the web sockets to work so it's a very flaky experience.

**Comment by saladin1980** [comment]
Work around does work and can confirm. Hope this is resolved at some point pain to get it working :(

**Comment by daveymg-nz** [comment]
@nittanygeek  great find, works perfectly, thanks for posting

**Comment by GitTimeraider** [comment]
Can add to this issue that when using the link on the server, it indeed gives you an agent you cannot install (Windows 11)
Workaround seems to work, but should be fixed regardless

**Comment by primetecabt** [comment]
> I found a temporary workaround. The button in Downloads is linking to https://redacted/API/ClientDownloads/WindowsInstaller, however after poking around the code a bit I found the actual Remotely_Installer.exe is located in https://redacted/Content/Remotely_Installer.exe
> 
> I was able to download the installer directly and then Right Click > Properties and unblock the file before I could run it.
> 
> Hope this helps.

Hi, My problem was resolved with the workaround provided

**Comment by bitbound** [comment]
Can you try again with the latest image?

**Comment by primetecabt** [comment]
> Can you try again with the latest image?

The executable worked, but the portal looked like this after the update.

Just correcting it, it returned to normal after clearing the browser cache.

![page](https://github.com/immense/Remotely/assets/145407448/f7aaba5b-4d33-4f0e-8ddf-a704c55b836a)

**Comment by saladin1980** [comment]
Worked on my side without an issue, although http - https issue with link but that is minor.    All is good and think this img update worked for my install.

**Comment by nittanygeek** [comment]
> Can you try again with the latest image?

Works here!  I'm still having branding issues, but I assume that's a separate issue.  From the logs:

```
[Error]	[v2023.09.11.1010]	[Process ID: 22652]	[Thread ID: 10]	[2023-09-19 13:21:06.697]	[Exception: ArgumentException]	[Immense.RemoteControl.Desktop.UI.ViewModels.ChatWindowViewModel]	Error applying branding.	Unable to load bitmap from provided data
   at Avalonia.Skia.ImmutableBitmap..ctor(Stream stream)
   at Avalonia.Skia.PlatformRenderInterface.LoadBitmap(Stream stream)
   at Avalonia.Media.Imaging.Bitmap..ctor(Stream stream)
   at Immense.RemoteControl.Desktop.UI.ViewModels.BrandedViewModelBase.<ApplyBrandingImpl>b__14_0() in C:\repos\Remotely\submodules\Immense.RemoteControl\Immense.RemoteControl.Desktop.UI\ViewModels\BrandedViewModelBase.cs:line 71
```

**Comment by tim1014** [comment]
> Can you try again with the latest image?

Hello, it worked as intended. Thanks for the fix.

**Comment by tim1014** [comment]
> Worked on my side without an issue, although http - https issue with link but that is minor. All is good and think this img update worked for my install.

add in nginx (if you are using): proxy_set_header X-Forwarded-Proto https;

**Comment by alexpi4** [comment]
Tip on how to download the Installer by "nittanygeek"

access your server by your installation address + port

Example

"http://192.168.1.211:5000/Content/Remotely_Installer.exe" and download

the executable.

Thanks for the Tip "nittanygeek"

After downloading the executable, follow the steps below

Installing manually on WINDOWS 8.1
If necessary, install [dot.net](http://dot.net/) 4.8, on some machines it may be necessary.

open CMD as administrator

copy "Remotely_Installer.exe" and "Remotely-Win-x64.zip" to c:\

open cmd as administrator

go to c:\

copy and paste the command below using the mouse button and include your "organizationid"

Remotely_Installer.exe -install -organizationid "xxxxx-xxxxx-xxxxx-xxxx" -serverurl "Youhost:port" -path "c:\Remotely-Win-x64.zip"

hit enter

and wait for the installation

:)

ALTERNATIVE COMMAND
Remotely_Installer.exe -install -organizationid "xxxxx-xxxxx-xxxxx-xxxx" -serverurl "http://192.168.1.211:5000/"
